src/modules/tfs/commentApp/commentApp.ts

```diff
@@ -82,6 +92,54 @@ export default class CommentApp extends LightningElement {
     this.firebase.auth().signInWithRedirect(provider);
   }
 
+  /**
+   * Handle a comment being liked by a user
+   * @param event The event with the comment to like
+   */
+  async handleLikeComment(event: CustomEvent) {
+    // get the comment to like and liking user from the event
+    const likedComment: Comment = event.detail.comment;
+    const likingUserId: string = event.detail.userId;
+
+    // like the comment as the current user
+    const likeResponse = await this._likeService.likeComment(
+      likedComment,
+      likingUserId
+    );
+
+    if (likeResponse.error) {
```

Comment:
Am I completely missing something here? This if block would get triggered if there was an error right? Not if no error like the comment suggests?

Overall, how interpret is that in the like component you set the like to liked before the firebase call fires so UI is snappy. Then when firebase returns you update the comment with what firebase returned.

Reply:
Good catch, that block should actually execute if their is no error. This will need to be fixed.

That's correct, since I set the comment to liked before the firebase call, this didn't cause issues.

Reply:
Ok cool! I guess thats the only change then I will merge it.

Reply:
Actually, the comment is technically incorrect. If there was an error when updating, we would want to replace the comment in order to undo setting the comment as "liked". I actually think it would be more consistent to reset the comment no matter what.

Reply:
I think we only need to reset if error though. Basically it's optimistic UI updating. It is an actual pattern.

Reply:
I am going to merge though. It looks good!